src/components/lib/crt/crt.c

```diff
@@ -468,6 +781,55 @@ crt_comp_alias_in(struct crt_comp *c, struct crt_comp *c_in, struct crt_comp_res
 	return 0;
 }
 
+
+
+int
+crt_sinv_create_shared(struct crt_sinv *sinv, char *name, struct crt_comp *server, struct crt_comp *client,
+		vaddr_t c_fn_addr, vaddr_t c_ucap_addr, vaddr_t s_fn_addr)
+{
+	struct cos_compinfo *cli;
+	struct cos_compinfo *srv;
+	unsigned int ucap_off;
+	struct usr_inv_cap *ucap;
+	
+	assert(sinv && name && server && client);
+
+	cli = cos_compinfo_get(client->comp_res);
+	srv = cos_compinfo_get(server->comp_res);
+
+	assert(crt_refcnt_alive(&server->refcnt) && crt_refcnt_alive(&client->refcnt));
+	crt_refcnt_take(&client->refcnt);
```

Comment:
I think that maybe we should be taking the reference counts when a component is referenced from a VAS NS? Not sure.